### PR TITLE
allow private key to be passed to interactive agent

### PR DESCRIPTION
**Pull request by wakamex**
attempt to account for 100% of liquidity in a pool

since some liquidity is deployed in the deployAndInitialize call. I found the minimum to be 20 in my test.

idea is for it to show up like this:
```
config = InteractiveHyperdrive.Config()
config.initial_liquidity = FixedPoint(11)
interactive_hyperdrive = InteractiveHyperdrive(chain, config)
privkey = chain.get_deployer_account_private_key()  # anvil account 0
pubkey = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
larry = interactive_hyperdrive.init_agent(base=FixedPoint(100_000), name="larry", private_key=privkey)
assert larry.wallet.address.hex().startswith(pubkey.lower())  # deployer public key
current_wallet = interactive_hyperdrive.get_current_wallet()
current_wallet
```
but no LP position shows up in `current_wallet`

### CODE REVIEW

**Comment by dpaiton** [comment]
This is not working, correct? I like the idea of any being able to initialize any agent0 Agent with a private key, so a general solution for this would be awesome.

**Review by dpaiton — APPROVED** [review]
Let's wait for @slundqui to approve before merging.

**Review by slundqui — APPROVED** [review]
LGTM

**Pull request merged**
